src/foam/u2/Element.js

```diff
@@ -1611,6 +1622,8 @@ foam.CLASS({
       this.onDetach(dao.listen(listener));
       listener.delegate.paint();
 
+      dao.sub('on', 'reset', listener.delegate.paint);
```

Comment:
I think this should be wrapped in this.onDetach() so that when the GUI is removed it doesn't keep getting events. Why is this calling listener.delegate.paint? Wouldn't listener be sufficient?

Reply:
Also, this seems like a problem with the DAO design. We are already doing a dao.listen() above. We shouldn't need to do both a listen() and sub(). @adamvy, why is this?

Reply:
@adamvy @kgrgreer Any update on this?

Reply:
Can you make a repro test case/demo that this fixes?  I have not had any problem with views not re-rendering when reset is called.

Reply:
Or if you can give me repro steps for nanoPayinc/NANOPAY#2537 I can try that as well.

Reply:
Repro steps for NANOPAY:

1. Go to "Invoices > Payable"
2. Click on a summary card at the top to filter invoices. It should work. Clicking again will deselect it and the table should update as expected.
3. Click on any invoice in the table.
4. Click the "Back" button in the upper left of the page.
5. Try filtering the table using the summary cards again, now it doesn't work.

Reply:
Look at https://github.com/nanoPayinc/NANOPAY/compare/b/NP-2537/fix-table-dao-update?expand=1 for a proper fix for that bug.